samples/nrf_desktop/src/modules/ble_adv.c

```diff
@@ -29,20 +44,71 @@ static const struct bt_data ad[] = {
 			  0x12, 0x18,	/* HID Service */
 #endif
 			  0x0f, 0x18),	/* Battery Service */
-	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
-};
 
-static const struct bt_data sd[] = {
+#if CONFIG_DESKTOP_BLE_SWIFT_PAIR
+	BT_DATA(BT_DATA_NAME_SHORTENED, DEVICE_NAME, DEVICE_NAME_SHORT_LEN),
+	BT_DATA_BYTES(BT_DATA_MANUFACTURER_DATA,
+			  0x06, 0x00,	/* Microsoft Vendor ID */
+			  0x03,		/* Microsoft Beacon ID */
+			  0x00,		/* Microsoft Beacon Sub Scenario */
+			  0x80),	/* Reserved RSSI Byte */
+#else
 	BT_DATA(BT_DATA_NAME_COMPLETE, DEVICE_NAME, DEVICE_NAME_LEN),
+#endif
 };
 
+static const struct bt_data sd[] = {};
+
 static bool bonds_initialized;
 static bool bonds_remove;
 
+#if CONFIG_DESKTOP_BLE_SWIFT_PAIR
```

Comment:
Please remove this preprocessor check and use `if (_IS_ENABLED(...))` where appropriate. Compiler will optimize this function and variables out.

Reply:
Done, used hidden Kconfig options for timeouts

Reply:
Done